.github/workflows/release-workflow-dispatcher.yml

```diff
@@ -0,0 +1,267 @@
+        required: true
+      release-branch:
+        required: true
+      helm-version:
+        required: true
+      app-version:
+        required: true
+  push:
+    branches:
+      - "release/**"
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+**"
+
+jobs:
+  create-release-branch:
+    if: github.event_name == 'workflow_dispatch'
+    runs-on: ubuntu-latest
+    outputs:
+      branch_exists: ${{ steps.check-branch.outputs.exists }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: develop # Adjust if your default branch is different
+          fetch-depth: 0
+          submodules: recursive
+      - name: Check if branch exists
+        id: check-branch
+        run: |
+          if git ls-remote --exit-code --heads origin ${{ github.event.inputs.release-branch }}; then
+            echo "Branch already exists: ${{ github.event.inputs.release-branch }}"
+            echo "exists=true" >> $GITHUB_ENV
+            echo "exists=true" >> $GITHUB_OUTPUT
+          else
+            echo "exists=false" >> $GITHUB_ENV
+            echo "exists=false" >> $GITHUB_OUTPUT
+          fi
+      - name: Create and push new branch
+        if: env.exists == 'false'
+        run: |
+          git checkout -b ${{ github.event.inputs.release-branch }}
+          git push origin ${{ github.event.inputs.release-branch }}
+
+  update-helm-chart:
+    needs: create-release-branch
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.release-branch }}
+          fetch-depth: 0
+          submodules: recursive
+
+      - uses: cachix/install-nix-action@v22
+      - name: Pre-populate nix-shell
+        run: |
+              export NIX_PATH=nixpkgs=$(jq '.nixpkgs.url' nix/sources.json -r)
+              echo "NIX_PATH=$NIX_PATH" >> $GITHUB_ENV
+              nix-shell --pure --run "echo" ./shell.nix
+    
+      - name: Test the chart version updater script
+        run: |
+              nix-shell --pure --run "./scripts/helm/test-update-chart-version.sh" ./shell.nix
+
+      - name: Modify chart version upon release branch creation.
+        run: |
+          branch_name=${{ github.event.inputs.release-branch }}
+          nix-shell --pure --run "./scripts/helm/update-chart-version.sh --branch $branch_name --type release --chart-version ${VERSION}-develop  --localpv-provisioner-version ${LOCALPV_PROVISIONER_VERSION} --zfs-localpv-version ${ZFS_LOCALPV_VERSION} --lvm-localpv-version ${LVM_LOCALPV_VERSION} --mayastor-version ${MAYASTOR_VERSION}" ./shell.nix
+        env:
+          VERSION: ${{ github.event.inputs.helm-version }}
+          LOCALPV_PROVISIONER_VERSION: ${{ github.event.inputs.localpv-provisioner-version }}
+          ZFS_LOCALPV_VERSION: ${{ github.event.inputs.zfs-localpv-version }}
+          LVM_LOCALPV_VERSION: ${{ github.event.inputs.lvm-localpv-version }}
```

Comment:
nit: let's call the envs LOCALPV_ZFS_VERSION, LOCALPV_LVM_VERSION, LOCALPV_HOSTPATH_VERSION

Reply:
Not fixing at the moment.